samples/highcharts/blog/blog-update-options/demo.js

```diff
@@ -0,0 +1,787 @@
+        if (t) {
+            clearInterval(t);
+        }
+        button.disabled = true;
+        demoChart.update({
+            chart: {
+                animation: false
+            }
+        });
+
+        /*
+            Physics object with three customizable variables
+            that change the behaviour of how the wheel behaves.
+            */
+
+        const springStrength = 0;
+        const dragValue = 0;
+        const springLength = 0;
+
+        const physics = {
+            force: 0,
+            angleVel: 0,
+            angle: 0,
+            prevAngle: 0,  // only used to calculate winner
+            strength: 0.003 + springStrength  / 10000, // tweakable
+            drag: 0.98 + dragValue / 1000,     // tweakable
+            threshold: 2 + springLength / 10,   // tweakable
+            targ: 0,
+            isActive: false
+        };
+        // How many degrees to spin for each iteration
+        let diff = 25 + Math.random() * 10,
+            startAngle = demoChart.options.plotOptions.item.startAngle;
+
+        const animationSpeed = Math.abs(parseFloat(animationSlider.value));
+        t = setInterval(() => { // Animation loop
+            if (!physics.isActive) {
+                startAngle += diff;
+                if (startAngle > 360) {
+                    startAngle -= 360;
+                }
+                diff *= 0.98;
+
+                demoChart.update({
+                    plotOptions: {
+                        pie: {
+                            startAngle: startAngle
+                        }
+                    }
+                });
+                // Transition to physics, and initialize the physics object.
+                if (diff < physics.threshold) {
+                    physics.isActive = true;
+                    /*
+                        Target should be the current angle.
+                        The wheel will spring back and forth with
+                        reference to this point.
+                    */
+                    physics.targ = startAngle;
+                    physics.angleVel = physics.threshold * 0.98;
+                    physics.angle = startAngle;
+
+                }
+            } else { // spring physics
+                physics.prevAngle = physics.angle;
+                physics.force = physics.targ - physics.angle;
+                physics.force *= physics.strength;
+                physics.angleVel *= physics.drag;
+                physics.angleVel += physics.force;
+                physics.angle += physics.angleVel;
+                demoChart.update(
+                    {
+                        plotOptions: {
+                            pie: {
+                                startAngle: physics.angle
+                            }
+                        }
+                    }
+                );
+
+            }
+            if (physics.prevAngle <= physics.angle) {
+                button.disabled = false;
+            }
```

Comment:
Instead of building the physics from scratch, we should use (and demonstrate how to use) our built-in animation for this. Plus it is much shorter. The approach is to define an easing function on the Math object, and call that from the animation options: https://jsfiddle.net/highcharts/cyLub6nm/

Reply:
Did you decide to remove the spinning pie altogether?